### PR TITLE
Enable localStorage and add caching

### PR DESCRIPTION
**Pull request by rusko124**
The discussion of the problems here is https://github.com/Screenly/screenly-ose/issues/806.

### CODE REVIEW

**Comment by vpetersson** [comment]
@sergey-borovkov can you take a quick look?

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -1,12 +1,25 @@
 #include <QWebFrame>
+#include <QStandardPaths>
```
Comment (sergey-borovkov) [inline src/mainwindow.cpp:2]: please sort includes alphabetically

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+    diskCache = new QNetworkDiskCache(this);
+    cacheLocation = QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
+
+    diskCache->setCacheDirectory(cacheLocation);
+    manager->setCache(diskCache);
+
+
     view = new View(this);
+    view -> page()->setNetworkAccessManager(manager);
+    view -> page()->settings()->setMaximumPagesInCache(1);
```
Comment (sergey-borovkov) [inline src/mainwindow.cpp]: what's the point of setting this? Users don't interact with the screens and this affects history.

File: src/mainwindow.cpp [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
 #include "mainwindow.h"
 #include "view.h"
 
 
 MainWindow::MainWindow() : QMainWindow()
 {
+    QWebSettings::globalSettings()->setAttribute(QWebSettings::LocalStorageEnabled, true);
+
+    manager = new QNetworkAccessManager(this);
+    diskCache = new QNetworkDiskCache(this);
```
Comment (sergey-borovkov) [inline src/mainwindow.cpp]: there is no point in making those variables a class member variables. They are not used anywhere and you can always get them back. Just declare them in the same function

**Comment by vpetersson** [comment]
@sergey-borovkov OK'd?

**Comment by sergey-borovkov** [comment]
yes.

**Comment by vpetersson** [comment]
Thanks, @rusko124! Good job.

**Pull request merged**